scripts/quality/recall_eval_cairo_auditor.py

```
@@ -0,0 +1,316 @@
+def _is_evaluable(finding: dict[str, Any]) -> bool:
+    snippet = str(finding.get("vulnerable_snippet", "")).strip().lower()
+    if len(snippet) < 25:
+        return False
+    return not any(marker in snippet for marker in PLACEHOLDER_MARKERS)
+
+
+def load_findings(findings_dir: Path) -> list[dict[str, Any]]:
+    findings: list[dict[str, Any]] = []
+    for fp in sorted(glob.glob(str(findings_dir / "*.findings.jsonl"))):
+        for line in Path(fp).read_text(encoding="utf-8").splitlines():
+            line = line.strip()
```

Ruff (0.15.15)

[warning] `for` loop variable `line` overwritten by assignment target

(PLW2901)